plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java

```diff
@@ -1860,6 +1860,12 @@ public void testMissingIndex()
         assertTableDoesNotExist("nonexistent_table");
     }
 
+    @Test
+    public void testRemoteQueryTableFunction()
+    {
+        assertQuerySucceeds("SELECT * FROM TABLE(\"" + getSession().getCatalog().orElseThrow() + "\".\"system\".\"remote_query\"(\"schema\" => 'tpch', \"index\" => 'nation', \"query\" => '{\"query\": {\"match\": {\"name\": \"ALGERIA\"}}}'))");
```

Comment:
It would be better to verify the result in my opinion and how about formatting like this? 
```java
        assertThat(
                "SELECT * FROM " +
                        "TABLE(\"" + getSession().getCatalog().orElseThrow() + "\".\"system\".\"remote_query\"(" +
                        " \"schema\" => 'tpch'," +
                        " \"index\" => 'nation'," +
                        " \"query\" => '{\"query\": {\"match\": {\"name\": \"ALGERIA\"}}}'" +
                        "))")
                .matches(...);
```

Also, I would recommend adding empt result & failure case.

Reply:
That's intentional. Pass-through queries in Elasticsearch return a single VARCHAR column as a result.

Reply:
@ebyhr I added more tests including no match and failure.